src/App/App.xaml.cs

```diff
@@ -168,13 +169,29 @@ private async Task CheckPasswordlessLoginRequestsAsync()
                 return;
             }
 
-
             var notification = await _stateService.GetPasswordlessLoginNotificationAsync();
             if (notification == null)
             {
                 return;
             }
 
+            var activeUserId = await _stateService.GetActiveUserIdAsync();
+            if (notification.UserId != activeUserId)
+            {
+                var notificationUserEmail = await _stateService.GetEmailAsync(activeUserId);
+                Device.BeginInvokeOnMainThread(async () =>
+                {
+                    var result = await _deviceActionService.DisplayAlertAsync(AppResources.LogInRequested, string.Format(AppResources.LoginAttemptFromXDoYouWantToSwitch, notificationUserEmail), AppResources.Cancel, AppResources.Ok);
+                    if (result == AppResources.Ok)
+                    {
+                        await _stateService.SetActiveUserAsync(notification.UserId);
+                        _messagingService.Send(AccountsManagerMessageCommands.SWITCHED_ACCOUNT);
+                    }
+                });
+
+                return;
+            }
```

Comment:
This method is getting big, please extract this into a different method and remember to use the awaitable method of `Device` => `await Device.InvokeOnMainThreadAsync` (in your case, in an exception happens in there it'll crash the app).

Reply:
1. Only the most request request is saved for now.
2. If the user cancels the automatic account switching, he can manually change the account and the request will pop.